tests/Controller/CreateBookingControllerTest.php

```diff
@@ -0,0 +1,781 @@
+
+        $resourceIdMock = 'DOKK1-Lokale-Test1@aarhus.dk';
+
+        $resource = new AAKResource();
+        $resource->setResourceName('DOKK1-Lokale-Test1');
+        $resource->setResourceDisplayName('DOKK1 Lokale Test1');
+        $resource->setResourceMail('DOKK1-Lokale-Test1@aarhus.dk');
+        $resource->setLocation('Dokk1');
+
+        $aakResourceRepositoryMock
+            ->expects($this->once())
+            ->method('findOneBy')
+            ->with(['resourceMail' => $resourceIdMock])
+            ->willReturn($resource)
+        ;
+
+        $container = self::getContainer();
+        $container->set(MetricsHelper::class, $metricMock);
+        $container->set(AAKResourceRepository::class, $aakResourceRepositoryMock);
+
+        $data = [
+            'abortIfAnyFail' => true,
+            'bookings' => [
+                [
+                    'subject' => 'Test Booking',
+                    'start' => '2004-02-26T15:00:00+00:00',
+                    'end' => '2004-02-26T15:30:00+00:00',
+                    'name' => 'Admin Jensen',
+                    'email' => 'admin_jensen@example.com',
+                    'resourceId' => 'DOKK1-Lokale-Test1@aarhus.dk',
+                    'clientBookingId' => '1234567890',
+                    'userId' => 'some_unqiue_user_id',
+                    'metaData' => [
+                        'data1' => 'example1',
+                        'data2' => 'example2',
+                    ],
+                ],
+            ],
+        ];
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage("An error occurred\n\nError validating booking. Aborting.");
+
+        $response = $client->request('POST', '/v1/bookings', [
+            'json' => $data,
+        ]);
+
+        $response->getContent();
+    }
+
+    public function testCreateBookingWithInvalidEmail(): void
+    {
+        $client = $this->getAuthenticatedClient();
+
+        $metricMock = $this->createMock(MetricsHelper::class);
+
+        $metricMock
+            ->expects($this->exactly(3))
+            ->method('incMethodTotal')
+            ->with(
+                $this->createCallback([
+                    'App\Security\ApiKeyAuthenticator::onAuthenticationSuccess',
+                    'App\Controller\CreateBookingController::__invoke',
+                    'App\Controller\CreateBookingController::__invoke',
+                ]),
+                $this->createCallback([
+                    'success',
+                    MetricsHelper::INVOKE,
+                    MetricsHelper::COMPLETE,
+                ]),
+            )
+        ;
+
+        $container = self::getContainer();
+        $container->set(MetricsHelper::class, $metricMock);
+
+        $data = [
+            'abortIfAnyFail' => false,
+            'bookings' => [
+                [
+                    'subject' => 'Test Booking',
+                    'start' => '2004-02-26T15:00:00+00:00',
```

Comment:
maybe the date should be valid in this case to avoid confusion
    'start' => '2004-02-26T15:00:00.010Z',